backend/infrahub/core/query/diff.py

```diff
@@ -519,7 +519,7 @@ async def query_init(self, db: InfrahubDatabase, **kwargs):
             MATCH (p)-[diff_rel]-(q)
             WHERE any(l in labels(p) WHERE l in ["Node", "Attribute", "Relationship"])
             AND %(diff_rel_filter)s
-            AND p.branch_support IN $branch_support
+            AND (p.branch_support IN $branch_support OR q.branch_support IN $branch_support)
```

Comment:
sometimes we need to include an agnostic node for the other side of a relationship even if we would normally leave all agnostic nodes out of the diff